### DISCUSSION TITLE
How to find out about dependencies of an `ApplicationModule`?

### DISCUSSION THREAD

**Discussion by nagkumar** [spring-projects/spring-modulith#234] [General]
This test code 

```
@Test
    void createApplicationModuleModel()
    {
	modules.forEach(System.out::println);
    }

```
gives output as

```
# Domain
> Logical name: domain
> Base package: com.teja.spring.modulith.full.domain
> Named interfaces:
  + NamedInterface: name=<<UNNAMED>>, types=[ c.t.s.m.f.d.OrderID ]
  + NamedInterface: name=events, types=[ c.t.s.m.f.d.e.OrderCompleted ]
> Spring beans: none

# Inventory
> Logical name: inventory
> Base package: com.teja.spring.modulith.full.inventory
> Spring beans:
  o ….InventoryInternal
  + ….InventoryMgmt
  o ….InventorySettings

# Orders
> Logical name: orders
> Base package: com.teja.spring.modulith.full.orders
> Spring beans:
  + ….OrderMgmt
  o ….internal.OrderInternal
```

however this output still missing what are the other modules it depends on.. is there a way to print that info too here..?

**Comment by nagkumar** [comment]
the option I see here is through spring actuator not from test case code

<img width="449" alt="image" src="https://github.com/spring-projects/spring-modulith/assets/332234/65e7f665-2a83-4bcc-ad56-0dae74c3e0ed">

**Comment by odrotbohm** [comment]
If you invoke `….toString()` on your `ApplicationModules` instance, each module also outputs its dependencies to other modules. Alternatively, invoke `ApplicationModule.toString(ApplicationModules)` individually.

You can inspect the details of the dependencies by calling `ApplicationModule.getDependencies(ApplicationModules)`. That'll return an `ApplicationModuleDependencies` which allows you to stream over `ApplicationModuleDependency` that exposes source, target and dependency type.
